### DISCUSSION TITLE
Can you enable secure connection in the exporters?

### DISCUSSION THREAD

**Discussion by scrhobbs** [NetApp/harvest#2069] [Q&A] [answered]
I want to enable https/tls in the exporters to make the connection secure, but everything I've tired does not work.
It's not specifically mentioned in the documentation for the exporter, only the service discovery. 
Is there any way to do it, or is it not supported?

**Comment by cgrinds** [comment]
hi @scrhobbs the Prometheus exporter does not currently support TLS. I'll turn this discussion into an issue and we'll add it.

**Comment by cgrinds** [comment]
Issue here #2070

**Reply by cgrinds** [reply]
hi @scrhobbs also wanted to share another option to achieve TLS exporters:
- change Prometheus exporter in harvest.yml to use `local_http_addr: localhost`
- setup an HTTPS reverse proxy via (Nginx, Caddy, Traefik, etc.) that presents an HTTPS endpoint to clients, and serves the localhost only Prometheus metrics

Depending on what you're trying to achieve, that sort of setup may help.

**Comment by cgrinds** [comment, marked as answer]
HI @scrhobbs [TLS enabled Prometheus exporters](https://netapp.github.io/harvest/nightly/prometheus-exporter/#prometheus-exporter-and-tls) are available nightly. If you want to try these out before the next release, you can grab the latest [nightly build](https://github.com/NetApp/harvest/releases/tag/nightly).
